retroshare-gui/src/gui/chat/ChatWidget.cpp

```diff
@@ -393,30 +430,99 @@ void ChatWidget::init(const ChatId &chat_id, const QString &title)
 				if (time(nullptr) <= historyIt->recvTime+2)
 					continue;
 
-				QString name;
-				if (chatId.isLobbyId() || chatId.isDistantChatId())
-				{
-					RsIdentityDetails details;
-					time_t start = time(nullptr);
-					while (!rsIdentity->getIdDetails(RsGxsId(historyIt->peerName), details))
-					{
-						std::this_thread::sleep_for(std::chrono::milliseconds(10));
-						if (time(nullptr)>start+2)
-						{
-							std::cerr << "ChatWidget History haven't found Id Details and have wait 1 sec for it." << std::endl;
-							break;
+				const std::string
+					notbeinghashableisagoodidea((const char*)historyIt->chatPeerId.toByteArray());
+				struct lookup_info& state = statecache[notbeinghashableisagoodidea];
+				if(!state.hasName) {
+					int tries;
+					for(tries=0;tries<4;++tries) {
+						assert(!state.hasName);
```

Comment:
generally we dont use assert() except when testing the code, but not to be merged in master. assert crashes the program without warning. Here it is preferable to gracefully quit the function. Use exceptions if you prefer, or a proper return after printing an error message.